networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2Peer.java

```diff
@@ -149,6 +151,8 @@ Optional<RequestKey> approveDataColumnSidecarsRequest(
 
   boolean approveRequest();
 
+  void recordStorageLimitHit();
```

Comment:
Not sure we should disclose reason on this layer. I'd use something more general like `pauseAllRequests()` or smth like that